src/server/game/AI/CreatureAI.h

```diff
@@ -228,6 +228,9 @@ class TC_GAME_API CreatureAI : public UnitAI
 
         static bool IsInBounds(CreatureBoundary const& boundary, Position const* who);
 
+        // Called every 3 Seconds to check if the creature needs to reposition itself
```

Comment:
On retail this time is about 1s, maybe less. Maybe it could be changed?

Reply:
My tests on retail have clearly shown me that it is more like 3 seconds. Maybe you went close to the creature when the timer has been started already

Reply:
Probably it was changed, but I don't know when. Maybe in TBC or WotLK.
Vanilla: https://youtu.be/SwLjJH8LXfc?t=566 https://youtu.be/SwLjJH8LXfc?t=161